File: CS/App_Code/CustomerDS.cs
using System;
using System.Web;
using System.Linq;
using NorthwindModel;
using System.Collections.Generic;

public class CustomerDS
{
    public List<Customer> GetCustomers(int startIndex, int maxRows)
    {
        using (NorthwindEntities entities = new NorthwindEntities())
        {
            return (from customer in entities.Customers
                    select customer)
                        .OrderBy(customer => customer.CustomerID)
                        .Skip(startIndex)
                        .Take(maxRows).ToList();
        }
    }

    public int GetCustomersCount()
    {
        using (NorthwindEntities entities = new NorthwindEntities())
        {
            return entities.Customers.Count();
        }
    }
}

File: VB/App_Code/CustomerDS.vb
Imports System.Linq
Imports NorthwindModel
Imports System.Collections.Generic

Public Class CustomerDS
    Public Function GetCustomers(startIndex As Integer, maxRows As Integer) As List(Of Customer)
        Using entities As New NorthwindEntities()
            Return (From customer In entities.Customers) _
                .OrderBy(Function(customer) customer.CustomerID) _
                .Skip(startIndex) _
                .Take(maxRows).ToList()
        End Using
    End Function

    Public Function GetCustomersCount() As Integer
        Using entities As New NorthwindEntities()
            Return entities.Customers.Count()
        End Using
    End Function
End Class
